braindecode/training/scoring.py

```diff
@@ -430,6 +430,8 @@ def predict_trials(module, dataset, return_targets=True):
     all_preds, all_ys, all_inds = [], [], []
     with torch.no_grad():
         for X, y, ind in loader:
+            device = next(module.parameters()).device
+            X = X.to(device)
```

Comment:
I could consider this as black magic. What if my data is already on the device? this generates so extra computation for nothing? Could say it's the responsability of the user to pass a dataset that is on the right device.

Reply:
I would disagree. The workflow is typically as follows:
A dataset is created, data is loaded onto CPU / not loaded at all.
A model is created on CPU and sent to GPU, if available / desired.
In the training loop, a batch is collected (loads data, if not already loaded) and put to the device compatible with the model.
Since we use skorch with the scikit-learn API `fit` and `predict`, the training loop is hidden and the device allocation should be / is handled by skorch.

`predict_trials` is now something special to braindecode, due to the cropped decoding. The function is __not__ called during training but only once after training has finished to create predictions on trial level instead of on compute window level as done during training. So we implement the data iteration ourselves and also the allocation to the appropriate device. The overhead of extra computations is also negligible.

Reply:
yes agree with Lukas, e.g. see
https://github.com/skorch-dev/skorch/blob/fe71e3d55a4ae5f5f94ef7bdfc00fca3b3fd267f/skorch/net.py#L208-L212